### PR TITLE
feat #34: optimize media referencing

### PR DESCRIPTION
**Pull request by kirlen**
this breaks out the media map so it can be used to populate media data URI's on demand (e.g. during rendering), instead of pre-populating the whole archive with media data before we store it:

- my 32.6Mb twitter.zip, which was being expanded to 81Mb, now is down to 50Mb. yay :)
- suggest we introduce a version mechanism, at whatever point we care about making sure uploaded archives remain supported (or not).

### CODE REVIEW

File: client/src/utils/store.js [hunk 1]
```diff
@@ -5,23 +5,23 @@ import { Zip } from "../process/Zip";
 import { Beejs } from "../process/Beejs";
 
 const reducerActions = (state = initialState, action) => {
-  console.log("action: ", action.type);
-  switch (action.type) {
+  const { type, payload } = action
```
[the remark this answers is not in the document]
Comment (kirlen) [reply client/src/utils/store.js:8]: destructuring, i hope, not destructing ;)

[the remark this answers is not in the document]
Comment (ameer-clara) [reply client/src/utils/store.js:8]: +1 for typo correction -10 for not providing a guess for a movie name

[the remark this answers is not in the document]
Comment (kirlen) [reply client/src/utils/store.js:8]: oh damn, you're gonna have to tell me, unless @codiak knows?

File: client/src/process/Zip.tsx [hunk 2]
```diff
@@ -93,38 +96,32 @@ export class Zip {
...
-          resolve(proccessedFile);
-        } catch (e) {
-          console.log("Error parsing media: ", e);
-          reject(proccessedFile);
+        processedFile.data = dataUrlProlog + content;
+        if (processedFile.name.includes("-")) {
+          processedFile.id = processedFile.name.substring(processedFile.name.lastIndexOf("-") + 1, processedFile.name.length);
         }
+        console.log("Add media: ", processedFile);
+        resolve(processedFile);
```
[the remark this answers is not in the document]
Comment (kirlen) [reply client/src/process/Zip.tsx:122]: so yeah, i scratched my head about your existing try/catch, which is inside the then block. i would maybe have expected your wanting to catch and handle exceptions thrown by the file.async call, but looks like instead you were more concerned with possible nullish exceptions that probably cropped up during development but look unlikely now. I'm actually gonna move 118-120 out of the then block, too, to make it clear that (unless i'm missing something), no exception handling is currently called for.

File: client/src/utils/store.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ameer-clara) [reply client/src/utils/store.js:8]: Dolemite Is My Name :)

File: client/src/process/Zip.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ameer-clara) [reply client/src/process/Zip.tsx:120]: 👍

**Review by ameer-clara — APPROVED** [review]

**Pull request merged**